util/edns0.go

```diff
@@ -0,0 +1,131 @@
+package util
+
+import "github.com/miekg/dns"
+
+// RemoveEdns0Record removes the OPT record from the Extra section of the given message.
+func RemoveEdns0Record(msg *dns.Msg) {
+	if msg == nil {
+		return
+	}
+
+	if msg.IsEdns0() == nil {
+		return
+	}
+
+	if len(msg.Extra) > 0 {
+		extra := make([]dns.RR, 0, len(msg.Extra))
+
+		for _, rr := range msg.Extra {
+			if rr.Header().Rrtype != dns.TypeOPT {
+				extra = append(extra, rr)
+			}
+		}
+
+		msg.Extra = extra
+	}
```

Comment:
Invert the `if`, modify in place, and short-circuit:
```suggestion
	if len(msg.Extra) == 0 {
		return
	}

	for i, rr := range msg.Extra {
		if rr.Header().Rrtype == dns.TypeOPT {
			msg.Extra = slices.Delete(msg.Extra, i, i+1)

			break
		}
	}
```